### ISSUE TITLE
API for subchart

### ISSUE THREAD

**Issue by JiaJian** [masayuki0812/c3#890]
Will there be an API for the subchart feature?

**Comment by JiaJian** [comment]
Oh wait, is it possible for me create my own API while extending c3js?
> https://github.com/masayuki0812/c3/releases/tag/0.3.0

**Comment by JiaJian** [comment]
I was messing around with the source, so far I'm able to create an API to show the subchart. The x-axis (with ticks) is showing but the data lines aren't showing...

**Comment by aendrew** [comment]
Sounds like a terrific pull request in the making! :smile: 

Could you post a link to your branch where you're doing this? Happy to take a look!

**Comment by JiaJian** [comment]
I left it as a gist [here](https://gist.github.com/JiaJian/62f510ad5795e516f407).

So far I've had luck generating the subchart and the brush by calling ``chart.subchart_show(true);`` after the c3.js ``chart`` has been generated. I realise there were more to just the subchart... there's also the brush. I couldn't figure how to hide the brush and update the svg defs if subchart_show is set to false.

**Comment by masayuki0812** [comment]
Hi, I fixed something about subchart, so I think now you can add `subchart.show` and `subchart.hide` API like this:
```
c3.chart.fn.subchart = function () {};
c3.chart.fn.subchart.show = function () {
    var $$ = this.internal;
    $$.config.subchart_show = true;

    $$.updateSizes();
    $$.updateTargetsForSubchart(chart.internal.data.targets);
    $$.showTargets();

    this.flush();
};
c3.chart.fn.subchart.hide = function () {
    var $$ = this.internal;
    $$.config.subchart_show = false;

    this.flush();
};
```
I'll release this fix as the next version v0.4.9, and I'm planning this API as a plugin in the future version.

**Comment by JiaJian** [comment]
Hey! Thank you so much! ``this.flush()`` didn't work as well, but ``$$.redraw()`` worked perfectly fine.
